src/main/java/fi/fmi/avi/model/swx/amd82/immutable/AirspaceVolumeImpl.java

```diff
@@ -40,12 +53,137 @@ public static Optional<AirspaceVolumeImpl> immutableCopyOf(final Optional<Airspa
+        return builder()
+                .setHorizontalProjection(geometry)
+                .withVerticalLimits(verticalLimits)
+                .build();
+    }
+
+    /**
+     * <p>
+     * Creates an airspace volume based on the location indicator and optional parameters.
+     * </p>
+     * <p>
+     * For {@link fi.fmi.avi.model.swx.amd79.SpaceWeatherRegion.SpaceWeatherLocation#DAYLIGHT_SIDE},
+     * computes the airspace volume using the sub-solar point and a static radius.
+     * For other location indicators with latitude bands, builds a polygon geometry using the
+     * location's latitude bounds and the provided longitude limits.
+     * </p>
+     *
+     * @param locationIndicator the location indicator
+     * @param analysisTime      analysis time (required only for DAYLIGHT_SIDE)
+     * @param minLongitude      minimum longitude (optional, defaults to -180)
+     * @param maxLongitude      maximum longitude (optional, defaults to 180)
+     * @param verticalLimits    vertical limit constraints
+     * @return the built airspace volume
+     */
+    public static AirspaceVolumeImpl fromLocationIndicator(final SpaceWeatherLocation locationIndicator,
+                                                           @Nullable final Instant analysisTime,
+                                                           @Nullable final Double minLongitude,
+                                                           @Nullable final Double maxLongitude,
+                                                           @Nullable final VerticalLimits verticalLimits) {
+        if (locationIndicator == SpaceWeatherLocation.DAYSIDE && analysisTime != null) {
+            return AirspaceVolumeImpl.forDaylightSide(analysisTime);
+        } else if (locationIndicator == SpaceWeatherLocation.NIGHTSIDE && analysisTime != null) {
+            // TODO
+            return AirspaceVolumeImpl.builder().build();
+        } else if (locationIndicator.getLatitudeBandMinCoordinate().isPresent()
+                && locationIndicator.getLatitudeBandMaxCoordinate().isPresent()) {
+            return AirspaceVolumeImpl.fromBounds(
+                    locationIndicator.getLatitudeBandMinCoordinate().get(),
+                    minLongitude != null ? minLongitude : -180d,
+                    locationIndicator.getLatitudeBandMaxCoordinate().get(),
+                    maxLongitude != null ? maxLongitude : 180d,
+                    verticalLimits != null ? verticalLimits : VerticalLimitsImpl.none());
+        }
+        throw new IllegalArgumentException("Unable to create AirspaceVolume for location indicator: " + locationIndicator);
+    }
+
     public abstract Builder toBuilder();
 
     public static class Builder extends AirspaceVolumeImpl_Builder {
-        @Deprecated
-        Builder() {
-        }
 
         public static Builder from(final AirspaceVolume value) {
```

Comment:
I think the `@Deprecated` annotation can be removed, but the empty constructor should be in place to retain the package-private visibility.